authors/apps/authentication/backends.py

```diff
@@ -53,3 +57,7 @@ def authenticate_credentials(self, request, token):
             raise exceptions.AuthenticationFailed('User has been deactivated')
         return user
```

Comment:
Custom authentication should return a tuple or None

Paul Kinuthia [11:13 PM]
To implement a custom authentication scheme, subclass BaseAuthentication and override the .authenticate(self, request) method. The method should return a *two-tuple of (user, auth)* if authentication succeeds, or None otherwise.

https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication

Reply:
Thank you @p8ul Ill definitely make the changes